### ISSUE TITLE
Improve BinaryCasReader/Writer

### ISSUE THREAD

**Issue by reckart** [dkpro/dkpro-core#390]
```
The support for the binary formats in the BinaryCasReader/Writer is still a bit spotty,
e.g. related to supporting the serialized formats currently covered by the SerializedCasReader/Writer
(the latter should be superseded by BinaryCasReader/Writer) and the ability to store
the type system separate from the CAS data and only once for a whole set of CASes -
this is in particular useful when many small texts are processes as the CAS metadata
causes significant bloat then when stored embedded in each binary file.
```

Original issue reported on code.google.com by `richard.eckart` on 2014-05-15 10:47:51
